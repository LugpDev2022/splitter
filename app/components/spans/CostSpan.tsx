interface CostSpanProps {
  quantity: number;
}

const CostSpan: React.FC<CostSpanProps> = ({ quantity }) => (
  <span className='text-teal-500 text-5xl font-bold'>
    ${quantity <= 0 ? '0.00' : quantity.toString()}
  </span>
);

export default CostSpan;
