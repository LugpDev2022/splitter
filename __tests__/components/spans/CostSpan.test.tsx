import CostSpan from '@/app/components/spans/CostSpan';
import { render, screen } from '@testing-library/react';

describe('Tests on <CostSpan />', () => {
  test('should match snapshot', () => {
    const { container } = render(<CostSpan quantity={3.33} />);
    expect(container).toMatchSnapshot();
  });

  test('should show the given quantity correctly', () => {
    render(<CostSpan quantity={3.33} />);
    const span = screen.getByText('$3.33');
    expect(span).toBeTruthy();
  });

  test('should show $0.00 when quantity is 0', () => {
    render(<CostSpan quantity={0} />);
    const span = screen.getByText('$0.00');
    expect(span).toBeTruthy();
  });
});
