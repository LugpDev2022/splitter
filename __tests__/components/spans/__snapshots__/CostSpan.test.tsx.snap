// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <CostSpan /> should match snapshot 1`] = `
<div>
  <span
    class="text-teal-500 text-5xl font-bold"
  >
    $
    3.33
  </span>
</div>
`;
